cpp/src/link_prediction/jaccard.cu

```diff
@@ -20,60 +20,67 @@
  * ---------------------------------------------------------------------------**/
 
 #include "utilities/graph_utils.cuh"
-#include "cugraph.h"
+#include "graph.hpp"
 #include "rmm_utils.h"
 #include "utilities/error_utils.h"
 
 namespace cugraph { 
 namespace detail {
+
   // Volume of neighboors (*weight_s)
-  template<bool weighted, typename IdxType, typename ValType>
+  template<bool weighted, typename vertex_t, typename edge_t, typename weight_t>
   __global__ void __launch_bounds__(CUDA_MAX_KERNEL_THREADS)
-  jaccard_row_sum(IdxType n,
-                  IdxType *csrPtr,
-                  IdxType *csrInd,
-                  ValType *v,
-                  ValType *work) {
-    IdxType row, start, end, length;
-    ValType sum;
+  jaccard_row_sum(vertex_t n,
+                  edge_t const *csrPtr,
+                  vertex_t const *csrInd,
+                  weight_t const *v,
+                  weight_t *work) {
+
+    vertex_t row;
```

Comment:
This is tedious, but we may better avoid leaving variables to have an arbitrary value.

vertex_t row{};

will help us to avoid some surprising bugs in the future.

And roughly speaking, initializing with {} is safer and more modern than initializing with () or =.

int i = std::numeric_limits<long>::max() generates a warning

but

int i{std::numeric_limits<long>::max()} fails to compile.

Reply:
Agree.  I've started using the {} initialization in areas where an initialization occurs.  Hadn't really thought about when a variable is left uninitialized.  

I'll try to keep this in mind going forward.